### DISCUSSION TITLE
Windows Java applications

### DISCUSSION THREAD

**Discussion by burrburr** [beehive-lab/TornadoVM#63] [Q&A]
Hello, I have a java program on my computer (.jar) that works well but is incredibly slow even though I have a 9th gen core I9. In my computer, I also have an RTX 4000 gpu and was wondering what steps I would need to take in order to get the application to run on my gpu if that is even possible. Additionally, I have tried to install tornado vm using the graal jdk 11 for fun and I have gotten stuck. When trying to compile I keep getting this error (see picture). Is there anything I can do to fix this? 
![image](https://user-images.githubusercontent.com/40464982/107869206-75a7ad80-6e59-11eb-9ff8-a2daf5061244.png)

**Comment by jjfumero** [comment]
Regarding the JAR file. Your applications need to be written using the TornadoVM API. Otherwise, it won't have access to the acceleration. TornadoVM does not auto-parallelize. Also, not all workloads are suitable for acceleration. We have a suite with many examples from different domains, such as Fintech, Machine Learning, Physics simulation, Astrophysics, etc. This suite can give you an idea of the type of workloads that TornadoVM can accelerate: https://github.com/beehive-lab/TornadoVM/tree/master/examples/src/main/java/uk/ac/manchester/tornado/examples/compute 

Regarding the error, are you following the steps described here (https://github.com/beehive-lab/TornadoVM/blob/master/assembly/src/docs/20_INSTALL_WINDOWS_WITH_GRAALVM.md) for Windows? It looks to me that there is an error in the `settings.xml` file from maven.

**Reply by burrburr** [reply]
I am following the steps from the website. I don't know what the error could be, I followed the install instructions on the maven website as well. Was I supposed to only extract, but not install?

**Reply by stratika** [reply]
Hi, you are supposed to download the compressed maven file (e.g. apache-maven-3.6.3-bin.tar.gz), and extract it to a directory. No need to install maven, the extract should be enough, as it will include the `bin` directory where the executable resides.

Then the key part for maven is step 6, here: https://github.com/beehive-lab/TornadoVM/blob/master/assembly/src/docs/20_INSTALL_WINDOWS_WITH_GRAALVM.md#msys2

What do you get if you try in the MSys2 terminal, the following?
```
which mvn
```

**Reply by burrburr** [reply]
This is the response I get 
![image](https://user-images.githubusercontent.com/40464982/107883771-f1881100-6ebe-11eb-9b25-315e45a45f73.png)

**Reply by stratika** [reply]
This looks correct. I am not sure what's the problem. Can you please post the `sources.env` file?

**Reply by stratika** [reply]
Can you try to add the maven binary in your ~/.bashrc file to see if it will work? Add this:
```
export PATH=/C/Maven/bin:$PATH
```

and then load it again with:
```
source ~/.bashrc
```

**Comment by jjfumero** [comment]
Your first error is related to the file `settings.xml` from maven. I don't think you have it. Could you please check you have it? 
It should be in your home directory `\home\denic\.m2\settings.xml`. We have specific content to fill in that file.

**Reply by stratika** [reply]
I think I got the problem, since I managed to reproduce it. The file `settings.xml` is automatically generated and configured during the building. It should not be configured by users. The confusion is because of the home directory, the ~/.m2 described in Step 5 is not the one that the `mvn` bash command in Step 6 aims to access.

You can try the following.
In Step 6, replace this:
`SETTINGS=\"${BASH_FILE%\\*\\*\\*}${HOME}/.m2/settings.xml\"` 
with this:
`SETTINGS=~/.m2/settings.xml`

This worked in my PC.

**Reply by jjfumero** [reply]
@burrburr , did this solve the problem?
